pkg/apis/serving/v1beta1/predictor.go

```diff
@@ -60,6 +63,24 @@ type PredictorSpec struct {
 	ComponentExtensionSpec `json:",inline"`
 }
 
+type ModelSpec struct {
+	// Framework of the model being served.
+	Framework Framework `json:"framework,omitempty"`
+
+	// Specific ClusterServingRuntime/ServingRuntime name to use for deployment.
+	// +optional
+	Runtime *string `json:"runtime,omitempty"`
+
+	PredictorExtensionSpec `json:",inline"`
+}
+
+type Framework struct {
+	// +required
+	Name string `json:"name"`
+	// +optional
+	Version *string `json:"version,omitempty"`
```

Comment:
This can cover multiple levels of granularity. Could just be major, major.minor, or even major.minor.patch. Currently the granularity has to exactly match a version specified in a ServingRuntime `supportedModelTypes` list, but logic can be added to add more flexibility so that an Isvc with a framework version of "1.4" could match a supportedModelType version of  just "1".